Explorer/Assets/DCL/UI/MainUIContainer/HideShowView.cs

```diff
@@ -0,0 +1,131 @@
+            showCancellationTokenSource = showCancellationTokenSource.SafeRestart();
+        }
+
+        private void OnPointerOnEnter()
+        {
+            if (forceOpen) return;
+
+            if (showing) { waitingToShow = false; }
+
+            if (showCancellationTokenSource.IsCancellationRequested) { waitingToShow = false; }
+
+            if (!showing && !waitingToShow)
+            {
+                waitingToShow = true;
+                showCancellationTokenSource = showCancellationTokenSource.SafeRestart();
+                WaitAndShow(showCancellationTokenSource.Token).Forget();
+            }
+
+            if (waitingToHide) { hideCancellationTokenSource.Cancel(); }
+        }
+
+        private void OnPointerOnExit()
+        {
+            if (forceOpen) return;
+
+            if (waitingToShow || showing) { showCancellationTokenSource.Cancel(); }
+
+            if (hideCancellationTokenSource.IsCancellationRequested) { waitingToHide = false; }
+
+            if (!waitingToHide && showing)
+            {
+                waitingToHide = true;
+                hideCancellationTokenSource = hideCancellationTokenSource.SafeRestart();
+                WaitAndHide(hideCancellationTokenSource.Token).Forget();
+            }
+        }
+
+        private async UniTask WaitAndHide(CancellationToken ct)
+        {
+            await UniTask.Delay(TimeSpan.FromSeconds(hideWaitTime), cancellationToken: ct);
+            waitingToHide = false;
+            if (ct.IsCancellationRequested) return;
+
+            await AnimateWidthAsync(hideLayoutWidth, ct);
+
+            if (ct.IsCancellationRequested) return;
+            showing = false;
+        }
+
+        private async UniTask WaitAndShow(CancellationToken ct)
+        {
+            await UniTask.Delay(TimeSpan.FromSeconds(showWaitTime), cancellationToken: ct);
+            waitingToShow = false;
+            if (ct.IsCancellationRequested) return;
+
+            await AnimateWidthAsync(showLayoutWidth, ct);
+
+            if (ct.IsCancellationRequested) return;
+            showing = true;
+        }
+
+        private async UniTask AnimateWidthAsync(float width, CancellationToken ct)
+        {
+            float startWidth = layoutElement.preferredWidth;
+            float endWidth = width;
+            var elapsedTime = 0f;
+
+            while (elapsedTime < animationTime)
+            {
+                if (ct.IsCancellationRequested)
+                {
+                    layoutElement.preferredWidth = startWidth;
+                    return;
+                }
+
+                elapsedTime += Time.deltaTime;
+                float t = Mathf.Clamp01(elapsedTime / animationTime);
+                layoutElement.preferredWidth = Mathf.Lerp(startWidth, endWidth, t);
+                await UniTask.Yield();
+            }
+
+            layoutElement.preferredWidth = endWidth;
+        }
+    }
```

Comment:
Can't we use DOTween for this animation? In this way we could have (and change it easily) a nice the showing/hidding effect.

Reply:
this one was removed xD but its the same case as the next one 👍